doc/src/manual/environment-variables.md

```diff
@@ -392,6 +392,23 @@ If set to anything besides `0`, then Julia's thread policy is consistent with
 running on a dedicated machine: the master thread is on proc 0, and threads are
 affinitized. Otherwise, Julia lets the operating system handle thread policy.
 
+## Garbage Collection
+
+### [`JULIA_HEAP_SIZE_HINT`](@id JULIA_HEAP_SIZE_HINT)
+
+Environment variable equivalent to the `--heap-size-hint` command line option.
+
+Forces garbage collection if memory usage is higher than the given value. The value may be specified as a number of bytes, optionally in units of:
+
+    - b  (bytes)
+    - k  (kibibytes)
+    - m  (mebibytes)
+    - g  (gibibytes)
+    - t  (tebibytes)
+    - %  (percentage of physical memory)
+
+which are not case sensitive. For example, `JULIA_HEAP_SIZE_HINT=1G` would provide a 1 GB heap size hint to the garbage collector.
```

Comment:
The units seem a bit inconsistent.. lowercase in the top list, uppercase in the example, then two-letter uppercase in the cli docs
https://github.com/JuliaLang/julia/blob/e1e7763e31d77c2b7c812a2fce4556bfd943f409/src/jloptions.c#L234-L237

Reply:
Just for reference – the current method in Julia, for whatever reason, is written as to not be case sensitive and actually only takes the first character – so, e.g., `5GB` would get parsed as `5g`. So technically these are all valid. For example:

```bash
julia --heap-size-hint=1Giga
```

will successfully set a 1 GB heap size hint. This is totally valid (though not documented, I guess?) on the current Julia.

I personally don't have a preference other than I agree consistency is always good. Wdyt? In retrospect I think the parsing should have required a specific case and number of characters. But too late for that now as all Julia back to 1.9 have this quirk.

Reply:
The inconsistency is also found in the existing docs. e.g., the current code is
```julia
    "                                               number of bytes, optionally in units of KB, MB, GB,\n"
    "                                               or TB, or as a percentage of physical memory with %.\n\n"
```
while the docs give the single-letter version: https://julialang.org/blog/2023/04/julia-1.9-highlights/#memory_usage_hint_for_the_gc_with_--heap-size-hint

Reply:
I would've thought matching here would make most sense https://docs.julialang.org/en/v1.12-dev/manual/command-line-interface/ so uppercase 2 chars. (Though I see [1.10](https://docs.julialang.org/en/v1.10/manual/command-line-interface/#cli) does give 'G` as an example.. but better to match current version)

Reply:
Ok will do the 2-char version, upper-case, everywhere.

Reply:
Hm, I take that back. It looks weird. Note that most unix tools, a single upper-case character is used. For example, `dd`, `rsync`, ulimit` - they all use K for kilobytes, G for gigabytes, M for megabytes. 

Even `ls` will display file sizes with a single character: `ls -l`.

Reply:
Ok I made everything single char in the docs. (Note that it's backwards compatible; it's just the docs were inconsistent before)